manager/rbac_manager.py

```diff
@@ -134,33 +136,33 @@ def format_permissions(res):
         return res
 
     @staticmethod
-    def need_permissions(perms: [RbacPermission], operator="or"):
+    def need_permissions(perms: [RbacPermission], operator="and"):
         """Decorator for checking if fetched users permissions are in given permissions for endpoint"""
         def decorator(func):
             def wrapper(*args, **kwargs):
                 if RbacManager.disable_rbac:
                     return func(*args, **kwargs)
 
-                rbac_perms = g.rbac_perms
-                # If vulnerability:*:* is in perms or rbac is disabled, needs to allow anything
-                if RbacRoutePermissions.VULNERABILITY_ANY in rbac_perms:
+                user_perms = g.rbac_perms
+                # If vulnerability:*:* and inventory:hosts:read is in perms or rbac is disabled, needs to allow anything
+                if RbacRoutePermissions.VULNERABILITY_ANY in user_perms and RbacRoutePermissions.INVENTORY_READ in user_perms:
                     return RbacManager.format_permissions(func(*args, **kwargs))
                 if operator == "or":
-                    # Is there any user permission which is in the set needed permissions in decorator?
-                    is_permitted = any(user_perm in perms for user_perm in rbac_perms)
+                    # For each default route permission, is some default permission in users permissions?
+                    is_permitted = any(default_perm in user_perms for default_perm in perms)
```

Comment:
Changed the way of checking perms to default route perm in user perms. The original way caused troubles on AND operator if user had some unknown permissions.